packages/kit/src/runtime/control.js

```diff
@@ -1,23 +1,21 @@
 export class HttpError {
-	// without these, things like `$page.error.stack` will error. we don't want to
-	// include a stack for these sorts of errors, but we also don't want red
-	// squigglies everywhere, so this feels like a not-terribile compromise
-	name = 'HttpError';
-
-	/** @type {void} */
-	stack = undefined;
-
 	/**
 	 * @param {number} status
-	 * @param {string | undefined} message
+	 * @param {{message: string} extends App.PageError ? (App.PageError | string | undefined) : App.PageError} body
 	 */
-	constructor(status, message) {
+	constructor(status, body) {
 		this.status = status;
-		this.message = message ?? `Error: ${status}`;
+		if (typeof body === 'string') {
+			this.body = { message: body };
```

Comment:
do we want to allow this, or should we require people to always pass the object?

Reply:
I think we should allow this for the common case where people don't really care about these things.